mailpoet/lib/AdminPages/Pages/NewsletterEditor.php

```diff
@@ -136,6 +144,8 @@ public function render() {
       'is_wc_transactional_email' => $newsletterId === $woocommerceTemplateId,
       'is_confirmation_email_template' => $newsletterId === $confirmationEmailTemplateId,
       'is_confirmation_email_customizer_enabled' => (bool)$this->settings->get('signup_confirmation.use_mailpoet_editor', false),
+      'product_categories' => $this->wpPostListLoader->getWooCommerceCategories(),
+      'products' => $this->wpPostListLoader->getProducts(),
```

Comment:
I have checked the code, and if I am not wrong, the result is empty when WC is not installed. So it looks safe to me.

Product and categories are here because the selection component reads them by the property `endpoint`. I used this approach because is also used in the dynamic segments form.

Reply:
> Can this component take all the products on a site?
I noticed you removed the pagination. This means this component will contain all the products available on the site.

I will continue this topic here because it is related. I am aware of it, and we use this approach in dynamic segments.
But I agree it is not the best solution for sites with a higher amount of products. I wanted to change my solution and use the property `remoteQuery`([#](https://github.com/mailpoet/mailpoet/blob/d116ff360b8d73f0ae09429ae0c365ed30a54def/mailpoet/assets/js/src/form/fields/selection.jsx#L150-L182)). Unfortunately, I cannot map the response from the endpoint `automatedLatestContent` without significant changes as it is [here](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/assets/js/src/newsletter_editor/blocks/coupon.ts#L434-L447).

I prefer to create a new ticket because this PR covers a lot of other changes. The previous solution contained this benefit, but I don't want a huge PR and want to unblock the rest of the tasks. Do you agree?

Reply:
Yeah, that is fine with me.